pkg/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -71,42 +72,25 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 						continue
 					}
 
-					if err := c.extractBOMWithTrivy(ctx, image); err != nil {
+					if err := c.extractBOMWithTrivy(ctx, image, ch); err != nil {
 						log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", image.Namespace, image.Name, err)
 					}
 				}
 			}
 		}
 	}()
 
-	return nil
-}
-
-func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata) error {
-	scanRequest := &docker.ScanRequest{
-		ImageMeta:    storedImage,
-		DockerClient: c.dockerUtil.RawClient(),
-	}
-
-	ch := make(chan sbom.ScanResult, 1)
-	if err := c.sbomScanner.Scan(scanRequest, c.scanOptions, ch); err != nil {
-		log.Errorf("Failed to trigger SBOM generation for docker: %s", err)
-		return err
-	}
-
 	go func() {
-		select {
-		case <-ctx.Done():
-		case result := <-ch:
+		for result := range ch {
 			if result.Error != nil {
 				log.Errorf("Failed to generate SBOM for docker: %s", result.Error)
-				return
+				continue
```

Comment:
should we requeue somewhere the image to try to rescan it later?

Reply:
It depends on the error. I think it deserves a separate PR.

Reply:
agree, could you just add a TODO here so we not forget.

Reply:
I added a comment everywhere we should add one